spec/models/manageiq/providers/amazon/cloud_manager/metrics_capture_spec.rb

```diff
@@ -58,18 +58,26 @@
   end
 
   context 'counters present' do
-    let(:vm_name) { 'i-091369dd8a8d470ab' }
     let(:ems) { FactoryBot.create(:ems_amazon_with_vcr_authentication, :provider_region => 'eu-central-1') }
+    let(:data) { vm.perf_collect_metrics('realtime').last.first.last }
```

Comment:
Maybe break this up into more variables so it is more descriptive?  the `.last.first.last` is really opaque

Reply:
Result of `perf_collect_metrics` looks like:
```
[
  { target.ems_ref => VIM_STYLE_COUNTERS },
  { 
    target.ems_ref => {
      timestamp => {
        counter_name => value,
        ...
      },
      timestamp... => { ... }
    }
  }
]
```
In the result we need to get: 
1. second (`== last`) hash from array, 
2. first (`== single`) key-value pair in that hash, 
3. and last element of that pair (`== value`).

Which will give us hash of `timestamp => { counter_name => value }` pairs.

I have no better idea to make it more descriptive, than to insert comment with that schema above.
Maybe you can propose something else?

Reply:
For readability I would recommend the following:

```ruby
let(:metrics) do
  # This is more obvious what is being returned than just seeing `.last`
  _vim_counters, amazon_counters = vm.perf_collect_metrics('realtime')
  amazon_counters
end
```

Then you can use the vm to dereference the hash instead of `.first.last`
```ruby
timestamps = metrics[vm.ems_ref] # This gets you the timestamps
```

This way it is more obvious what the structure of the data is, trust me if you just see `get_metrics.last.first.last` it is not obvious what is going on.

Reply:
changed there: https://github.com/ManageIQ/manageiq-providers-amazon/pull/525